src/engine/enginebuffer.cpp

```diff
@@ -1178,48 +1189,47 @@ void EngineBuffer::processSyncRequests() {
     }
 }
 
-void EngineBuffer::processSeek() {
-    // We need to read position just after reading seekType, to ensure that we read
-    // the matching poition to seek_typ or a position from a new seek just queued from an other thread
-    // the later case is ok, because we will process the new seek in the next call anyway.
-    SeekRequest seekType =
-            static_cast<SeekRequest>(m_iSeekQueued.fetchAndStoreRelease(NO_SEEK));
-    double position = m_queuedPosition.getValue();
+void EngineBuffer::processSeek(bool paused) {
+    // We need to read position just after reading seekType, to ensure that we
+    // read the matching position to seek_typ or a position from a new (second)
+    // seek just queued from an other thread
+    // The later case is ok, because we will process the new seek in the next
+    // call anyway again.
+
+    SeekRequests seekType = static_cast<SeekRequest>(
+            m_iSeekQueued.fetchAndStoreRelease(SEEK_NONE));
+    double position = m_queuedSeekPosition.getValue();
+
+    // Add SEEK_PHASE bit, if any
+    seekType |= static_cast<SeekRequests>(
+            m_iSeekPhaseQueued.fetchAndStoreRelease(SEEK_NONE));
```

Comment:
this interface is now confusing because m_iSeekQueued could still be set to SEEK_PHASE
It looks like SEEK_PHASE should be removed, and m_iSeekPhaseQueued should only be set to true or false